### PR TITLE
Chore expose store parameters

### PR DESCRIPTION
**Pull request by jeroiraz**

### CODE REVIEW

File: embedded/cache/lru_cache.go [hunk 1]
```diff
@@ -49,6 +51,17 @@ func NewLRUCache(size int) (*LRUCache, error) {
 	}, nil
 }
 
+func (c *LRUCache) Resize(size int) {
```
Comment (byo) [inline embedded/cache/lru_cache.go:54]: I don't see this method used in the code other than tests, is this a preparation for future changes about live updates of parameters?

File: embedded/cache/lru_cache.go [hunk 2]
```diff
@@ -49,6 +51,17 @@ func NewLRUCache(size int) (*LRUCache, error) {
...
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	for size < c.lruList.Len() {
+		c.evict()
```
[the remark this answers is not in the document]
Comment (jeroiraz) [reply embedded/cache/lru_cache.go:59]: good point. probably is not safe to rely on the data is in the cache so much. However the cache is meant to be a private component and allowing to be resized could be decided by the component itself. e.g. it's safe to evict nodes from the btree cache and it won't heart to reduce cache size. On the other hand, appendable interface is not exposing the cache nor a way to dynamically resize it

File: embedded/cache/lru_cache.go (hunk 1 above)
Comment (jeroiraz) [reply embedded/cache/lru_cache.go:54]: haven't labeled the PR as WIP,  yes, it's under preparation

**Comment by jeroiraz** [comment]
Covered by #1112

**Pull request closed without merging**